lib/aws-manager.js

```diff
@@ -231,70 +254,24 @@ class AwsManager {
   }
 
   /**
-   * Handle SpotRequests that we consider to have stalled.  For now, this means
-   * just cancel them.  In future this will do nifty things like maintaining
-   * state about which type/region/zone combinations are not working well right
-   * now
-   */
-  async handleStalledRequests (spotReqs) {
-    for (let region of Object.keys(spotReqs)) {
-      try {
-        await this.killCancel(region, [], spotReqs[region].map(sr => { //eslint-disable-line no-loop-func
-          debug('killing stalled spot request ' + sr.SpotInstanceRequestId);
-          return sr.SpotInstanceRequestId;
-        }));
-        debug('killed and cancelled stalled spot requests in %s', region);
-      } catch (err) {
-        debug('error killing and cancelling stalled spot requests in %s', region);
-        debug(err);
-        if (err.stack) {
-          debug(err.stack);
-        }
-        throw err;
-      }
-    }
-  }
-
-  /**
-   * Find out which availability zones are available
-   */
-  _filterAZ (res) {
-    let zones = {};
-
-    for (let region of this.ec2.regions) {
-      // This check might be redundant, but being extra-cautious
-      if (!res[region]) {
-        zones[region] = [];
-      } else {
-        zones[region] = res[region].AvailabilityZones.map(x => x.ZoneName); // eslint-disable-line no-loop-func
-      }
-    }
-
-    return zones;
-  }
-
-  /**
    * Get rid of the keys that I don't care about
    */
   _findMaxPrices (res, zones) {
-    // region -> type -> zone
+    // type -> zone
     let pricing = {};
 
-    for (let region of this.ec2.regions) {
-      pricing[region] = {};
-      for (let pricePoint of res[region].SpotPriceHistory) {
-        let type = pricePoint.InstanceType;
-        let price = parseFloat(pricePoint.SpotPrice, 10);
-        let zone = pricePoint.AvailabilityZone;
-
-        // Remember that we only want to consider available zones
-        if (zones[region].includes(zone)) {
-          if (!pricing[region][type]) {
-            pricing[region][type] = {};
-          }
-          if (!pricing[region][type][zone] || pricing[region][type][zone] < price) {
-            pricing[region][type][zone] = price;
-          }
+    for (let pricePoint of res.SpotPriceHistory) {
+      let type = pricePoint.InstanceType;
+      let price = parseFloat(pricePoint.SpotPrice, 10);
+      let zone = pricePoint.AvailabilityZone;
+
+      // Remember that we only want to consider available zones
+      if (zones.includes(zone)) {
+        if (!pricing[type]) {
+          pricing[type] = {};
+        }
+        if (!pricing[type][zone] || pricing[type][zone] < price) {
+          pricing[type][zone] = price;
```

Comment:
I think this is a comment for another time, but you would be a lot better off with:
```js
pricing[type] = [
 {zone: ..., price: ...},
]
```
Well, it depends on how it is used later... if you're looking for the cheapest zone (which I suspect, then having an array makes it easier to some form of max function... I'm sure lodash has one to select max object from a list...